modules/gltf/src/glb-arrow-loader.ts

```diff
@@ -0,0 +1,135 @@
+import {GLBLoader} from './glb-loader';
+
+/** GLB Arrow loader options */
+export type GLBArrowLoaderOptions = LoaderOptions & {
+  io?: PlatformIO;
+};
+
+export type ArrowTableTransformList = [ArrowTable, Matrix4][];
+
+/**
+ * GLB Loader -
+ * GLB is the binary container format for GLTF
+ */
+export const GLBArrowLoader = {
+  ...GLBLoader,
+  dataType: null as unknown as ArrowTableTransformList,
+  batchType: null as never,
+  worker: false,
+  parse,
+  parseSync: undefined
+} as const satisfies LoaderWithParser<ArrowTableTransformList, never, GLBArrowLoaderOptions>;
+
+async function parse(
+  arrayBuffer: ArrayBuffer,
+  options?: GLBArrowLoaderOptions
+): Promise<ArrowTableTransformList> {
+  const io = options?.io || new WebIO().registerExtensions(KHRONOS_EXTENSIONS);
+  const document = await io.readBinary(new Uint8Array(arrayBuffer));
+
+  // Unclear how represent indexed, instanced, or normalized meshes as
+  // ArrowTable. Convert to simpler representations for now.
+  await document.transform(unweld(), uninstance(), dequantize());
+
+  const scene = document.getRoot().getDefaultScene() || document.getRoot().listScenes()[0];
+  const meshList: ArrowTableTransformList = [];
+
+  // Traverse the default scene, creating a list of mesh primitives and their
+  // corresponding scene transforms.
+  scene.traverse((node: Node) => {
+    if (node.getMesh()) {
+      const matrix = new Matrix4(node.getWorldMatrix());
+      for (const prim of node.getMesh()!.listPrimitives()) {
+        meshList.push([convertPrimitiveToArrowTable(prim), matrix]);
+      }
+    }
+  });
+
+  return meshList;
+}
+
+/**
+ * Encodes a glTF Transform Primitive as an ArrowTable. Currently ignores
+ * materials, morph targets, and extras.
+ */
+function convertPrimitiveToArrowTable(prim: Primitive): ArrowTable {
+  const fields: Field[] = [];
+  const arrowFields: arrow.Field[] = [];
+  const arrowAttributes: arrow.Data[] = [];
+
+  let vertexCount = -1;
+
+  for (const name of prim.listSemantics()) {
+    const attribute = prim.getAttribute(name)!;
+    const type = componentTypeToDataType(attribute.getComponentType());
+
+    const field: Field = {name, type};
+    const arrowField = deserializeArrowField(field);
+    const arrowAttribute = accessorToArrowListData(attribute);
+
+    fields.push(field);
+    arrowFields.push(arrowField);
+    arrowAttributes.push(arrowAttribute);
+
+    if (vertexCount <= 0) {
+      vertexCount = attribute.getCount();
+    }
+  }
+
+  const metadata: SchemaMetadata = {};
+  const schema: Schema = {fields, metadata};
+
+  const arrowSchema = new arrow.Schema(arrowFields);
+  const arrowStruct = new arrow.Struct(arrowFields);
+  const arrowData = new arrow.Data(arrowStruct, 0, vertexCount, 0, undefined, arrowAttributes);
+  const arrowRecordBatch = new arrow.RecordBatch(arrowSchema, arrowData);
+  const arrowTable = new arrow.Table([arrowRecordBatch]);
```

Comment:
If we happen to know that a scene contains ~100 mesh that are similar[^1], possibly they could be returned as a single Table, with a separate table of transforms?

[^1]: Where 'similar' might mean ... sharing a material? ... same vertex attribute types? ... same node transform?